dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/RenameReferenceVisitor.java

```diff
@@ -0,0 +1,230 @@
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.openapi20.merge;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.microprofile.openapi.models.OpenAPI;
+import org.eclipse.microprofile.openapi.models.Operation;
+import org.eclipse.microprofile.openapi.models.Reference;
+import org.eclipse.microprofile.openapi.models.callbacks.Callback;
+import org.eclipse.microprofile.openapi.models.examples.Example;
+import org.eclipse.microprofile.openapi.models.headers.Header;
+import org.eclipse.microprofile.openapi.models.links.Link;
+import org.eclipse.microprofile.openapi.models.media.Schema;
+import org.eclipse.microprofile.openapi.models.parameters.Parameter;
+import org.eclipse.microprofile.openapi.models.parameters.RequestBody;
+import org.eclipse.microprofile.openapi.models.responses.APIResponse;
+import org.eclipse.microprofile.openapi.models.security.SecurityRequirement;
+import org.eclipse.microprofile.openapi.models.security.SecurityScheme;
+
+import io.openliberty.microprofile.openapi20.merge.MergeProcessor.NameType;
+import io.openliberty.microprofile.openapi20.merge.MergeProcessor.RenameHolder;
+import io.openliberty.microprofile.openapi20.utils.OpenAPIModelVisitor;
+import io.openliberty.microprofile.openapi20.utils.OpenAPIModelWalker.Context;
+
+/**
+ * Visitor which updates references within an OpenAPI model after things have been renamed
+ * <p>
+ * Updates:
+ * <ul>
+ * <li>references to objects in {@link OpenAPI#getComponents()}
+ * <li>references from links to operationIds
+ * <li>references to tags from operations
+ * </ul>
+ */
+public class RenameReferenceVisitor implements OpenAPIModelVisitor {
+
+    private final RenameHolder renameHolder;
+
+    /**
+     * Creates a RenameReferenceVisitor
+     * 
+     * @param renameHolder the {@code RenameHolder} holding the renames for the model
+     */
+    public RenameReferenceVisitor(RenameHolder renameHolder) {
+        this.renameHolder = renameHolder;
+    }
+
+    private void updateReference(Reference<?> referrer, NameType nameType, String prefix) {
+        String reference = referrer.getRef();
+        if (reference != null && reference.startsWith(prefix)) {
+            String oldName = reference.split("/")[3];
+            String newName = renameHolder.lookupName(nameType, oldName);
+            if (!oldName.equals(newName)) {
+                referrer.setRef(prefix + newName);
+            }
+        }
+    }
+
+    @Override
+    public Example visitExample(Context context, Example example) {
+        updateReference(example, NameType.EXAMPLES, "#/components/examples/");
```

Comment:
Constant?

Reply:
all of the strings that start `#/` e.g. `#/components/links/`

Reply:
As you suggested, I've added these strings to the `NameTypes` enum (which has also moved to its own file).